x-pack/solutions/observability/plugins/streams/server/routes/streams/processing/simulation_handler.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core/server';
+import { isEmpty } from 'lodash';
+import { FieldDefinitionConfig, RecursiveRecord } from '@kbn/streams-schema';
+import { calculateObjectDiff, flattenObject } from '@kbn/object-utils';
+import { SimulationFailedError } from '../../../lib/streams/errors/simulation_failed_error';
+import { DetectedMappingFailureError } from '../../../lib/streams/errors/detected_mapping_failure_error';
+import { NonAdditiveProcessorError } from '../../../lib/streams/errors/non_additive_processor_error';
+import { formatToIngestProcessors } from '../../../lib/streams/helpers/processing';
+import { ProcessingSimulateParams } from './route';
+
+export const prepareSimulationBody = (params: ProcessingSimulateParams) => {
+  const { path, body } = params;
+  // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
I'm fine with disabling this linting rule entirely for this plugin, but if we don't, maybe we can just use `detected_fields: detectedFields`?